### DISCUSSION TITLE
How to dynamic type for styled component?

### DISCUSSION THREAD

**Discussion by thienvu23** [tamagui/tamagui#2054] [General]
### Current Behavior

import { FlatList } from 'react-native';
const FlatListStyled = styled(FlatList);

I get an ItemType is an unknown type

### Expected Behavior

I want using a dynamic type for ItemType of FlatList

### Tamagui Version

```markdown
1.78.0
```

### Platform (Web, iOS, Android)

```markdown
react-native expo
```

### Reproduction

```markdown
Create a dynamic type for styled FlatList component
```

**Comment by natew** [comment]
Still an issue in the latest?

**Comment by natew** [comment]
Seems to work for me, not sure what the ask is

**Comment by natew** [comment]
Oh it's to type the renderItem? Good point, this may be useful in the future as a feature. Converting to discussion.

**Comment by byteab** [comment]
as a workaround we can do this

```const FlatListStyled = styled(FlatList<DataType>);```

**Comment by thienvu23** [comment]
@TheEhsanSarshar This might be problematic if I create a custom FlatList. How can I cast my Dynamic Type into the FlatList's DataType? It looks like this
<img width="698" alt="image" src="https://github.com/tamagui/tamagui/assets/91444569/86d1687b-cb09-474e-aa5c-f88edab5aff9">

**Reply by byteab** [reply]
you can do like so
```js
const createFlatList = <DataType,>() => {
     return styled(FlatList<DataType>)
}
const MyList = createFlatList<MyListDataType>()
```

**Reply by thienvu23** [reply]
Thank bro 🫡
